test/api/junction_detection_test.cpp

```diff
@@ -7,29 +7,38 @@
 
 // TEST(group1, fasta_out_empty)
 // {
-//     std::string expected{"Reference\tchr9\t70103073\tForward\tReference\tchr9\t70103147\tForward\tm13802/6999/CCS\n"};
+//     std::string expected{"Reference\tchr9\t70103073\tForward\tReference\tchr9\t70103147\tForward\tm13802/6999/CCS\t1\n"};
 //     testing::internal::CaptureStdout();
 //     detect_junctions_in_alignment_file(DATADIR"simulated.minimap2.hg19.coordsorted_cutoff.sam", "");
 //     std::string std_cout = testing::internal::GetCapturedStdout();
 //     EXPECT_EQ(expected, std_cout);
 // }
 
+// Explanation for the stings:
+// Reference\tm2257/8161/CCS\t41972616\tForward\tRead \t0\t2294\tForward\tchr21
+// INS from Primary Read - Sequence Type: Reference; Sequence Name: m2257/8161/CCS; Position: 41972616; Orientation: Reverse
+//                         Sequence Type: Read; Sequence Name: 0; Position: 3975; Orientation: Reverse
+//                         Chromosome: chr21
+// Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\tm41327/11677/CCS
+// BND from SA Tag - Sequence Type: Reference; Chromosome: chr22; Position: 17458417; Orientation: Forward
+//                   Sequence Type: Reference; Chromosome: chr21; Position: 41972615; Orientation: Forward
+//                   Sequence Name: m41327/11677/CCS
+
 TEST(junction_detection, fasta_out_not_empty)
 {
     std::string expected{
-        "Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\tm41327/11677/CCS\n"
-        "Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\tm21263/13017/CCS\n"
-        "Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\tm38637/7161/CCS\n"
-        "Reference\tm2257/8161/CCS\t41972616\tForward\tRead \t0\t2294\tForward\tchr21\n"
-        "Reference\tm2257/8161/CCS\t41972616\tReverse\tRead \t0\t3975\tReverse\tchr21\n"
+        "Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\tm41327/11677/CCS\t1\n"
+        "Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\tm21263/13017/CCS\t1\n"
+        "Reference\tm2257/8161/CCS\t41972616\tForward\tRead \t0\t2294\tForward\tchr21\t2\n"
+        "Reference\tm2257/8161/CCS\t41972616\tReverse\tRead \t0\t3975\tReverse\tchr21\t1\n"
```

Comment:
I stumbled about these strings that seem to be incorrect (as explained in my comment). We need to find out where they come from.

Reply:
I had a look and probably there is a mixup in the passing of chromosome and read in 
`retrieve_aligned_segments` or `analyze_cigar`.
I opened an issue for this which we should address soon: https://github.com/seqan/iGenVar/issues/52